### DISCUSSION TITLE
Updating jsonb array of objects

### DISCUSSION THREAD

**Discussion by springerjanek** [supabase/supabase#15234] [Questions] [answered]
I have row in my db thats type of jsonb array and looks like this:
`[
  {
    "name": "test2",
    "dueDate": "2000/11/11"
  }
]`

I want to know how can I update the array by adding a new object so it would contain 2 objects and so on.
I tried with `.update({
        row: [
          {
            name: "test3",
            dueDate: "2000/12/12",
          },
        ],
      })` passing array and a new object but instead of adding it it just replaced it.

**Comment by GaryAustin1** [comment, marked as answer]
Not from the API.  update is on complete columns.   Postgres though has lots of json and array operators so you could use an rpc call to do the update in SQL.

**Comment by zigzag-way** [comment]
Nobody knows? Is is possible to update JSONB fields with supabase-js?

**Reply by GaryAustin1** [reply]
It is not,as I said.  You have to use an rpc call and do it with SQL.

**Reply by zigzag-way** [reply]
Mama mia... I always face such issues. At first glance you have an easy to use supabase-js interface but then you got broke on the very first table with jsonb )))
